llvm/lib/Target/AMDGPU/AMDGPUISelLowering.cpp

```diff
@@ -59,6 +59,12 @@ unsigned AMDGPUTargetLowering::numBitsSigned(SDValue Op, SelectionDAG &DAG) {
 AMDGPUTargetLowering::AMDGPUTargetLowering(const TargetMachine &TM,
                                            const AMDGPUSubtarget &STI)
     : TargetLowering(TM), Subtarget(&STI) {
+  // Always lower memset, memcpy, and memmove intrinsics to load/store
+  // instructions, rather then generating calls to memset, mempcy or memmove.
+  MaxStoresPerMemset = MaxStoresPerMemsetOptSize = ~0U;
```

Comment:
Can you document where the `~0U` comes from?
Is it a magic value recognized in this pass or somewhere else?

Could we eventually have a constant variable for that value so its source is clear?

Reply:
If you wanted to be fancy you could use `std::numeric_limits<typeof(MaxStoresPerMemset)>::max()`

Reply:
> This is not a magic number, it is basically inf. These numbers are used as a threshold that determines whether the three functions will be lowered to load/store or lib call. Since we don't have lib call anyway, we need to lower them in all cases.

I see, `~0U` works for me then. Maybe just add a small comment on top explaining that this is the libcall lowering threshold and it shouldn't be reachable?